raysort/main_push.py

```diff
@@ -185,13 +185,18 @@ def reduce(self) -> list[PartInfo]:
             for r, merge_out in enumerate(merge_results):
                 if len(tasks_in_flight) >= self.cfg.reduce_parallelism:
                     _, tasks_in_flight = ray.wait(tasks_in_flight, fetch_local=False)
+
                 ref = final_merge.options(**ray_utils.current_node_aff()).remote(
-                    self.cfg, self.worker_id, r, *merge_out
+                    self.cfg, self.worker_id, r, list(merge_out)
                 )
                 merge_results[r, :] = None
-                tasks_in_flight.append(ref)
-                results.append(ref)
-            return ray.get(results)
+                if isinstance(ref, list):
```

Comment:
Can we just make `final_merge` always return a list (even if it's a singleton) and remove this check

Reply:
i think this was the list unwrapping issue i mentioned last meeting. I don't see anywhere that isn't returning a list, but for some reason if i dont add this check it fails.

Reply:
Ok, let's leave this as is.